plugins/module_utils/wafv2.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+try:
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+
+def wafv2_list_web_acls(wafv2, scope, fail_json_aws, Nextmarker=None):
+    # there is currently no paginator for wafv2
+    req_obj = {
+        'Scope': scope,
+        'Limit': 100
+    }
+    if Nextmarker:
+        req_obj['NextMarker'] = Nextmarker
+
+    try:
+        response = wafv2.list_web_acls(**req_obj)
+    except (BotoCoreError, ClientError) as e:
+        fail_json_aws(e, msg="Failed to list wafv2 web acl.")
+
+    if response.get('NextMarker'):
+        response['WebACLs'] += wafv2_list_web_acls(wafv2, scope, fail_json_aws, Nextmarker=response.get('NextMarker')).get('WebACLs')
+    return response
+
+
+def wafv2_list_rule_groups(wafv2, scope, fail_json_aws, Nextmarker=None):
```

Comment:
I'm a bit annoyed by the similarity with `wafv2_list_web_acls` but it sounds like we cannot really merge both without creating something worst :-).

Reply:
I hope the two function become obsolete, once `wafv2` will get support for paginator - one day.